src/frontend/js/components/Icon/index.tsx

```diff
@@ -0,0 +1,58 @@
+interface IconProps {
+  name: Icon;
+  /**
+   * Optional title, will be announced by screen readers
+   * and set as `title` attribute for mouse users
+   *
+   * If omitted, the icon is considered decorative and
+   * ignored by screen readers
+   */
+  title?: string;
+  className?: string;
+}
+
+export const Icon = ({ name, title, className = '' }: IconProps) => {
+  return (
+    <svg
+      {...{
+        className: `icon ${className}`,
+        ...(title
+          ? {
+              role: 'img',
+              'aria-label': title,
+              title,
+            }
+          : {
+              'aria-hidden': true,
+            }),
+      }}
+    >
+      <use href={`#${name}`} />
+    </svg>
+  );
+};
+
+// icons from src/richie/apps/core/templates/richie/icons.html
+type Icon =
```

Comment:
The idea is to restrict the `name` attribute to known available icons. Here it's just a manual copy of all icons in the `icons.html` template. So, with this merged, one should be careful in the future when adding an icon in `icons.html`: adding the icon id in the type here will also be necessary.

Didn't bother trying to automate the generation of this type as I figured it was rather OK to require this tiny new thing to do when adding an icon, I'll you decide.